### PR TITLE
misc/open_pdks: add gf180mcuc variant

### PR DESCRIPTION
**Pull request by proppy**
This add a new subpackages including the `C` variant (5-metal backend stack) of https://github.com/google/gf180mcu-pdk.

In order to support fully OpenLane, additional patches from [efabless's fork of open_pdks](https://github.com/efabless/open_pdks) should probably be included:
https://github.com/RTimothyEdwards/open_pdks/compare/master...efabless:open_pdks:master

### CODE REVIEW

**Comment by proppy** [comment]
rebased and ready for review.

@QuantamHD @mithro

File: misc/open_pdks/use-sources-from-conda.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+-GF180MCU_SC_9T5V0_PATH = @GF180MCU_FD_SC_MCU9T5V0_PATH@
+-GF180MCU_SRAM_PATH = @GF180MCU_FD_IP_SRAM_PATH@
++GF180MCU_PR_PATH = ../gf180mcu-pdk/gf180mcu_fd_pr
++GF180MCU_IO_PATH = ../gf180mcu-pdk/gf180mcu_fd_io
++GF180MCU_SC_7T5V0_PATH = ../gf180mcu-pdk/gf180mcu_fd_sc_mcu7t5v0
++GF180MCU_SC_9T5V0_PATH = ../gf180mcu-pdk/gf180mcu_fd_sc_mcu9t5v0
++GF180MCU_SRAM_PATH = ../gf180mcu-pdk/gf180mcu_fd_ip_sram
+ 
+ # Path to GF180MCU library sources
+ PDK_URL = https://github.com/google
```
Comment (QuantamHD) [inline misc/open_pdks/use-sources-from-conda.patch:21]: Is this correct?

Comment (proppy) [reply misc/open_pdks/use-sources-from-conda.patch:21]: yes, this get cloned by the recipe `sources:` and get passed to `./configure`, let me make this more explicit.

Comment (proppy) [reply misc/open_pdks/use-sources-from-conda.patch:21]: Done with https://github.com/hdl/conda-eda/pull/238/commits/004d21b47706e4d198fef4d977bf4d54cc5966c0

**Comment by proppy** [comment]
@PiotrZierhoffer @ajelinski it seems that conda-build-prepare doesn't like `open_pdks` subpackages: it seems to drop the `outputs` and goes into an infinite recursion:
https://pipelines.actions.githubusercontent.com/serviceHosts/b1561d4a-c200-469d-96ed-a5c4b56e2e69/_apis/pipelines/1/runs/1607/signedlogcontent/100?urlExpires=2022-09-08T08%3A17%3A19.4576764Z&urlSigningMethod=HMACV1&urlSignature=JeSfPNM5T6Ps7%2B1lthwjlO2ui%2FjMONeXD804i%2FY0vBI%3D
```
2022-09-08T07:23:42.3113866Z RecursionError: maximum recursion depth exceeded in comparison
```

Is there a way to workaround this, should we create two separate packages instead?

**Comment by proppy** [comment]
@QuantamHD @PiotrZierhoffer @ajelinski PTAL, splitted the packages to workaround https://github.com/litex-hub/conda-build-prepare/issues/7

**Comment by QuantamHD** [comment]
Lgtm, but I have no merge access

**Comment by proppy** [comment]
>  Lgtm, but I have no merge access

@QuantamHD There was a typo in the workflows that prevented it from running, fixed with https://github.com/hdl/conda-eda/pull/238/commits/5580f692f7eef16b34edb97e6f92719b1badaef8

**Comment by proppy** [comment]
@PiotrZierhoffer @ajelinski do you have some clue about why `open_pdks-linux-gf180mcuc` is being skipped?

**Comment by proppy** [comment]
@QuantamHD @mithro @hzeller can you take a look?

**Comment by QuantamHD** [comment]
What do you need help with in this case?

**Comment by proppy** [comment]
@QuantamHD reviewing this PR, so that we can merge and get a packages published by the CI

**Comment by proppy** [comment]
@ajelinski is there a workaround? (maybe https://github.com/marketplace/actions/maximize-build-disk-space? or dedicated workers?)

I'd like to be able to leverage published build of the package to ease gf180mcuc testing.

**Comment by PiotrZierhoffer** [comment]
@proppy We discussed this internally and having a GCP custom runner seems to be the best option. We'll see into setting this up

**Comment by proppy** [comment]
@PiotrZierhoffer any news on getting this setup>?

**Comment by proppy** [comment]
@xobs fyi: rebased this, let's see if it builds now that we have custom runners!

**Comment by xobs** [comment]
Looks reasonable to me. I look forward to trying this on some of my designs using the Conda workflow.

**Pull request merged**